validationparser/interceptor.go

```diff
@@ -0,0 +1,57 @@
+package validator
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"github.com/infobloxopen/atlas-app-toolkit/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+type validator interface {
+	Validate() error
+}
+
+// UnaryServerInterceptor returns a new unary server interceptor that validates incoming messages.
+//
+// Invalid messages will be rejected with `InvalidArgument` before reaching any userspace handlers.
+func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if v, ok := req.(validator); ok {
+			if err := v.Validate(); err != nil {
+				return nil, err
```

Comment:
return nil, ValidationHelper(err)

Reply:
We don't want to return the Validation Error here because the mapper only works with lyft validations. If there are other validations that exist we want to handle them differently we would want to add a mapper for each type of validation. That is why we want to just return the error so that our mapper can correctly handle it.

Reply:
got it